command/agent/template/template.go

```diff
@@ -255,16 +267,24 @@ func newRunnerConfig(sc *ServerConfig, templates ctconfig.TemplateConfigs) (*ctc
 		ServerName: pointerutil.StringPtr(""),
 	}
 
-	if strings.HasPrefix(sc.VaultConf.Address, "https") || sc.VaultConf.CACert != "" {
-		skipVerify := sc.VaultConf.TLSSkipVerify
+	if strings.HasPrefix(*conf.Vault.Address, "https") || sc.AgentConfig.Vault.CACert != "" {
```

Comment:
Since we're now pointing the CT's vault config to the agent's listener, shouldn't we be checking and using values from `sc.AgentConfig.Listerners[0]` (e.g. `sc.AgentConfig.Listerners[0].TLSCertFile` instead of `c.AgentConfig.Vault.CACert`) and fall back to using values from `sc.AgentConfig.Vault` if the cache stanza is not present?

Reply:
We actually override the address at 253 so this should be correct, however, this logic should be reworked a bit.

Reply:
Reworked, thanks for pointing this out!